tests/listengine.js

```diff
@@ -3345,6 +3345,45 @@ describe( 'ListEngine', () => {
 			expect( getViewData( editor.editing.view, { withoutSelection: true } ) )
 				.to.equal( '<ul><li>Foo<span></span><ul><li>Xxx</li><li>Yyy</li></ul></li></ul>' );
 		} );
+
+		describe( 'remove converter should properly handle ui elements', () => {
+			let uiElement;
+
+			beforeEach( () => {
+				editor.setData( '<ul><li>Foo</li><li>Bar</li></ul>' );
+				uiElement = new ViewUIElement( 'span' );
+			} );
+
+			it( 'ui element before <ul>', () => {
+				// Append ui element before <ul>.
+				viewRoot.insertChildren( 0, [ uiElement ] );
+
+				modelDoc.batch().remove( modelRoot.getChild( 0 ) );
```

Comment:
Uh... it took me 30s to understand this test. It wasn't clear for me until I understood that `getChild( 0 )` is `<listItem>Foo</listItem>`. Please comment such lines.